pkg/imageengine/buildah/from.go

```diff
@@ -0,0 +1,243 @@
+		return "", err
+	}
+
+	return builder.ContainerID, builder.Save()
+}
+
+func (engine *Engine) CreateContainer(opts *options.FromOptions) (string, error) {
+	wrapper := &fromFlagsWrapper{
+		FromAndBudResults: &buildahcli.FromAndBudResults{},
+		UserNSResults:     &buildahcli.UserNSResults{},
+		NameSpaceResults:  &buildahcli.NameSpaceResults{},
+	}
+
+	flags := engine.Flags()
+	fromAndBudFlags, err := buildahcli.GetFromAndBudFlags(wrapper.FromAndBudResults, wrapper.UserNSResults, wrapper.NameSpaceResults)
+	if err != nil {
+		return "", err
+	}
+
+	flags.AddFlagSet(&fromAndBudFlags)
+
+	err = engine.migrateFlags2BuildahFrom(opts)
+	if err != nil {
+		return "", err
+	}
+
+	return engine.createContainerFromImage(opts)
+}
+
+func (engine *Engine) migrateFlags2BuildahFrom(opts *options.FromOptions) error {
+	return nil
+}
+
+func onBuild(builder *buildah.Builder, quiet bool) error {
+	ctr := 0
+	for _, onBuildSpec := range builder.OnBuild() {
+		ctr = ctr + 1
+		commands := strings.Split(onBuildSpec, " ")
+		command := strings.ToUpper(commands[0])
+		args := commands[1:]
+		if !quiet {
+			fmt.Fprintf(os.Stderr, "STEP %d: %s\n", ctr, onBuildSpec)
+		}
+		switch command {
+		case "ADD":
+		case "COPY":
+			dest := ""
+			size := len(args)
+			if size > 1 {
+				dest = args[size-1]
+				args = args[:size-1]
+			}
+			if err := builder.Add(dest, command == "ADD", buildah.AddAndCopyOptions{}, args...); err != nil {
+				return err
+			}
+		case "ANNOTATION":
+			annotation := strings.SplitN(args[0], "=", 2)
+			if len(annotation) > 1 {
+				builder.SetAnnotation(annotation[0], annotation[1])
+			} else {
+				builder.UnsetAnnotation(annotation[0])
+			}
+		case "CMD":
+			builder.SetCmd(args)
+		case "ENV":
+			env := strings.SplitN(args[0], "=", 2)
+			if len(env) > 1 {
+				builder.SetEnv(env[0], env[1])
+			} else {
+				builder.UnsetEnv(env[0])
+			}
+		case "ENTRYPOINT":
+			builder.SetEntrypoint(args)
+		case "EXPOSE":
+			builder.SetPort(strings.Join(args, " "))
+		case "HOSTNAME":
+			builder.SetHostname(strings.Join(args, " "))
+		case "LABEL":
+			label := strings.SplitN(args[0], "=", 2)
+			if len(label) > 1 {
+				builder.SetLabel(label[0], label[1])
+			} else {
+				builder.UnsetLabel(label[0])
+			}
+		case "MAINTAINER":
+			builder.SetMaintainer(strings.Join(args, " "))
+		case "ONBUILD":
+			builder.SetOnBuild(strings.Join(args, " "))
+		case "RUN":
+			var stdout io.Writer
+			if quiet {
+				stdout = ioutil.Discard
+			}
+			if err := builder.Run(args, buildah.RunOptions{Stdout: stdout}); err != nil {
+				return err
+			}
+		case "SHELL":
+			builder.SetShell(args)
+		case "STOPSIGNAL":
+			builder.SetStopSignal(strings.Join(args, " "))
+		case "USER":
+			builder.SetUser(strings.Join(args, " "))
+		case "VOLUME":
+			builder.AddVolume(strings.Join(args, " "))
+		case "WORKINGDIR":
+			builder.SetWorkDir(strings.Join(args, " "))
+		default:
+			logrus.Errorf("unknown OnBuild command %q; ignored", onBuildSpec)
```

Comment:
I think we should return an error, since unexpected command is input. And we should consider it as illegal behaviour.